src/main/java/de/dennisguse/opentracks/io/file/exporter/KmlTrackWriter.java

```diff
@@ -322,32 +324,29 @@ private void writePlacemark(String name, String category, String description, St
     }
 
     /**
-     * Writes a photo overlay.
-     *  @param name        the name
-     * @param category    the category
-     * @param description the description
-     * @param location    the location
-     * @param photoUrl    the photo url
-     * @param heading     the heading
+     * Writes a photo overlay from waypoint.
+     *
+     * @param waypoint Waypoint object.
+     * @param heading  the heading.
      */
-    private void writePhotoOverlay(String name, String category, String description, Location location, String photoUrl, float heading) {
-        if (location != null && exportTrackDetail) {
+    private void writePhotoOverlay(Waypoint waypoint, float heading) {
+        if (waypoint.getLocation() != null && exportTrackDetail) {
```

Comment:
Waypoint.getLocation() cannot be null as it is required during creation.
Why should it here be different?